pkg/registry/generic/etcd/etcd.go

```diff
@@ -127,6 +131,28 @@ func NamespaceKeyFunc(ctx api.Context, prefix string, name string) (string, erro
 	return key, nil
 }
 
+// NoNamespaceKeyRootFunc is the default function for constructing etcd paths to root-scoped resource directories.
+func NoNamespaceKeyRootFunc(ctx api.Context, prefix string) (string, error) {
```

Comment:
Did we have a situation where we had non-namespace objects calling a NamespaceKeyFunc instead of the NoNamespaceKeyFunc?  If so, it seems it like it may be better to protect that a NamespaceKeyFunc is not configured to work with a non-namespaced object during configuration of generic etc object.